code/game/objects/structures/crates_lockers/closets/secure/cargo.dm

```diff
@@ -207,39 +207,60 @@
 				new /obj/item/ammo_magazine/speed_loader_rifle_75(src)
 				new /obj/item/ammo_magazine/speed_loader_rifle_75(src)
 		if("EGUN")
-			if(prob(50))
-				new /obj/item/gun/energy/cog(src)
-				new /obj/item/cell/medium(src)
-				new /obj/item/cell/medium(src)
-				new /obj/item/cell/medium(src)
-			else
-				new /obj/item/gun/energy/lasercore(src)
-				new /obj/item/cell/medium(src)
-				new /obj/item/cell/medium(src)
-				new /obj/item/cell/medium(src)
+			switch(rand(1,100))
+				if(1 to 35) //35% chance Cog
+					new /obj/item/gun/energy/cog(src)
+					new /obj/item/cell/medium(src)
+					new /obj/item/cell/medium(src)
+					new /obj/item/cell/medium(src)
+				if(36 to 65) //30% chance lasercore
+					new /obj/item/gun/energy/lasercore(src)
+					new /obj/item/cell/medium(src)
+					new /obj/item/cell/medium(src)
+					new /obj/item/cell/medium(src)
+				if(66 to 95) //30% chance Lightfall
+					new /obj/item/gun/energy/laser(src)
+					new /obj/item/cell/medium(src)
+					new /obj/item/cell/medium(src)
+					new /obj/item/cell/medium(src)
+				if(96-100) //5% chance Concillium, you only get two cells because they're large
```

Comment:
is this correct? Its unlike the others

Reply:
It is correct, it's coded like this to permit more than two options on a coinflip without resorting to another pickweight. I tested it, the various things within this list *can* spawn.